backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ConflictException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { UsersService } from 'src/users/users.service';
+import { SignUpDto } from './dto/sign-up.dto';
+import { SignInDto } from './dto/sign-in.dto';
+import * as bcrypt from 'bcrypt';
+import { User } from '@prisma/client';
+
+@Injectable()
+export class AuthService {
+  private EXPIRATION_TIME = '1m';
+  private ISSUER = 'Bankme';
+  private AUDIENCE = 'users';
+
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly usersService: UsersService,
+  ) {}
+
+  async signUp(signUpDto: SignUpDto) {
+    const { login } = signUpDto;
+    const user = await this.usersService.findLogin(login);
+
+    if (user !== null) {
+      throw new ConflictException('Login alread in use!');
+    }
```

Comment:
duas coisas sobre esse if:
- podemos utilizar o ! ao invez de fazer uma validação ativa com "não null", fica mais legivel e mais flexivel
- quando você retorna ao user que `Login already in use`, você pode estar abrindo uma brecha de segurança.